### PR TITLE
Add pay api template for swift

### PR DESCRIPTION
**Pull request by telip007**
Changes: 
- Added Pay API Swift template
- Versioning: Base URL is giving in the template yml and can be changed in the client. Base URLs should be without version. Version will be injected from a request when creating the url request. Version can be changed in request object.

### CODE REVIEW

**Comment by EmDee** [comment]
@telip007 Do we really need duplicate templates here?

**Comment by telip007** [comment]
> @telip007 Do we really need duplicate templates here?

@EmDee Well not exactly duplicates the template yml has different names etc... Also we will need to rename the classes `APIClient` etc to their respective API, for example: `POIAPIClient`, `PayAPIClient` etc...

**Comment by telip007** [comment]
what I try to look for in the process if we can somehow include some files in the cloudsdk which won't need to be generated every time and are equal in both apis, so we don't need to generate the twice or more with different names...

**Comment by telip007** [comment]
Also this is not the final structure yet, I just created a template and generated the API to see where we get errors in generation. So it will most likely include less duplicates to none than there are now.

**Comment by telip007** [comment]
@EmDee, unfortunately, it does not work to extract some common template files and put them in a "parent template" to be used by other child templates. This means in our template folders we have some duplicated files, however, we can generate them in the same place for all templates, so for example helper files like `AnyCodable.swift` can be generated into a `Common` folder in the parent folder by both (i.e. pay and poi) templates. So this would lead to folder structure like the following: `./Api/Common`, `./Api/Pay/Generated` and `./Api/POI/Generated` for example.

**Comment by EmDee** [comment]
> @EmDee, unfortunately, it does not work to extract some common template files and put them in a "parent template" to be used by other child templates. This means in our template folders we have some duplicated files, however, we can generate them in the same place for all templates, so for example helper files like `AnyCodable.swift` can be generated into a `Common` folder in the parent folder by both (i.e. pay and poi) templates. So this would lead to folder structure like the following: `./Api/Common`, `./Api/Pay/Generated` and `./Api/POI/Generated` for example.

I like the idea of splitting the generated files into `Common`, `Pay`, `POI` etc. However, I think it would be better to have a `Commons` template that generates all the common files and only add the templates that differ in the API templates.

The structure would be `API/Generated/Common`, `Api/Pay/Generated` and `Api/POI/Generated` for example.

**Comment by telip007** [comment]
@EmDee this should be ready now. Removed the tests for now for swaggen, but we should add them later on I think.

File: Templates/Swift/Pay/Includes/Enum.stencil [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{% if description %}
```
Comment (EmDee) [inline Templates/Swift/Pay/Includes/Enum.stencil:1]: The `Enum.stencil` in `Poi` is identical to this one, also there is also one in the `Common` folder, so this is probably not needed anymore?

File: Templates/Swift/Pay/Includes/Header.stencil [hunk 2]
```diff
@@ -0,0 +1,4 @@
+//
```
Comment (EmDee) [inline Templates/Swift/Pay/Includes/Header.stencil:1]: Common stencil?

File: Templates/Swift/POI/Includes/Header.stencil [hunk 3]
```diff
@@ -0,0 +1,4 @@
+//
```
Comment (EmDee) [inline Templates/Swift/POI/Includes/Header.stencil:1]: Common?

File: Templates/Swift/Pay/Includes/Model.stencil [hunk 4]
```diff
@@ -0,0 +1,175 @@
+{% if description %}
```
Comment (EmDee) [inline Templates/Swift/Pay/Includes/Model.stencil:1]: Common?

File: Templates/Swift/Pay/README.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
+# {{ options.name }}
+
+This is an api generated from a OpenAPI 3.0 spec with [SwagGen](https://github.com/yonaskolb/SwagGen)
```
Comment (EmDee) [inline Templates/Swift/Pay/README.md:3]: SwagGen link should be updated.

File: Templates/Swift/Pay/Sources/API.swift [hunk 6]
```diff
@@ -0,0 +1,55 @@
+{% include "Includes/Header.stencil" %}
```
Comment (EmDee) [inline Templates/Swift/Pay/Sources/API.swift:1]: This file is also almost identical to the one in `Poi`, except for the server name. Can we combine these, e.g. in a stencil file?

**Comment by EmDee** [comment]
@telip007 It still looks like we have a lot of duplicated code in the individual template folder. It looks like usually they only differentiate in small prefixes or names. Can we make use of template inheritance of the stencil files here somehow: https://stencil.fuller.li/en/latest/templates.html#template-inheritance ?

**Comment by EmDee** [comment]
@telip007 Dang. I miss the "diff since last push" feature of GitLab here...

**Comment by telip007** [comment]
@EmDee I removed every duplicated code :D

**Comment by telip007** [comment]
@EmDee added commits from the original repo and the removal and validation of query items inside paths.

**Pull request merged**